### PR TITLE
Setup mTLS gateway and access exposed workload with certificate tutorials

### PR DESCRIPTION
**Pull request by dariusztutaj**
**Description**

Changes proposed in this pull request:

- Add a new tutorial with complete steps on how setup mTLS gateway
- Add a new tutorial with complete steps on how to expose and secure a deployed workload with a certificate 
- Enhancements in the docs to adapt to the new changes

**Related issue(s)**
#15078

### CODE REVIEW

File: docs/03-tutorials/00-api-exposure/apix-10-expose-and-secure-workload-with-certificate.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+   apiVersion: networking.istio.io/v1alpha3
+   kind: VirtualService
+   metadata:
+     name: httpbin-vs
+     namespace: ${NAMESPACE}
+   spec:
+     hosts:
+     - "httpbin-vs.${DOMAIN_TO_EXPOSE_WORKLOADS}"
+     gateways:
+     - ${GATEWAY}
```
Comment (videlov) [inline docs/03-tutorials/00-api-exposure/apix-10-expose-and-secure-workload-with-certificate.md:47]:
```suggestion
     - ${MTLS_GATEWAY_NAME}
```

File: docs/03-tutorials/00-api-exposure/apix-10-expose-and-secure-workload-with-certificate.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+   ```bash
+   cat <<EOF | kubectl apply -f - 
+   apiVersion: networking.istio.io/v1alpha3
+   kind: VirtualService
+   metadata:
+     name: function-vs
+     namespace: ${NAMESPACE}
+   spec:
+     hosts:
+     - "function-example.${DOMAIN_TO_EXPOSE_WORKLOADS}"
```
Comment (videlov) [inline docs/03-tutorials/00-api-exposure/apix-10-expose-and-secure-workload-with-certificate.md:98]:
```suggestion
     - "function-vs.${DOMAIN_TO_EXPOSE_WORKLOADS}"
```

File: docs/03-tutorials/00-api-exposure/apix-10-expose-and-secure-workload-with-certificate.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+   kind: AuthorizationPolicy
+   metadata:
+     name: test-authz-policy
+     namespace: ${NAMESPACE}
+   spec:
+     action: ALLOW
+     rules:
+     - to:
+       - operation:
+           hosts: ["function-example.${DOMAIN_TO_EXPOSE_WORKLOADS}"]
```
Comment (videlov) [inline docs/03-tutorials/00-api-exposure/apix-10-expose-and-secure-workload-with-certificate.md:128]:
```suggestion
           hosts: ["function-vs.${DOMAIN_TO_EXPOSE_WORKLOADS}"]
```

File: docs/03-tutorials/00-security/sec-02-mtls-selfsign-client-certicate.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+   ```bash
+   export CLIENT_CERT_CN={COMMON_NAME}
+   export CLIENT_CERT_ORG={ORGANIZATION}
+   export CLIENT_CERT_CRT_FILE=${CLIENT_CERT_CN}.crt
+   export CLIENT_CERT_CSR_FILE=${CLIENT_CERT_CN}.csr
+   export CLIENT_CERT_KEY_FILE=${CLIENT_CERT_CN}.key
+   ```
+
+2. Generate a Client Root CA and a Client certificate:
```
Comment (videlov) [inline docs/03-tutorials/00-security/sec-02-mtls-selfsign-client-certicate.md:37]:
```suggestion
2. Generate a Client certificate and sign it with the Root CA:
```

File: docs/03-tutorials/00-api-exposure/apix-10-expose-and-secure-workload-with-certificate.md [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+  <summary>
+  Call the secured Function
+  </summary>
+
+Send a `GET` request with a token that has the `read` scope to the Function:
+
+   ```shell
+   curl --key ${CLIENT_CERT_KEY_FILE} \
+        --cert ${CLIENT_CERT_CRT_FILE} \
+        --cacert ${CLIENT_ROOT_CA_CRT_FILE} \ -ik -X GET https://function-example.$DOMAIN_TO_EXPOSE_WORKLOADS/function
```
Comment (videlov) [inline docs/03-tutorials/00-api-exposure/apix-10-expose-and-secure-workload-with-certificate.md:166]:
```suggestion
        --cacert ${CLIENT_ROOT_CA_CRT_FILE} \
        -ik -X GET https://function-vs.$DOMAIN_TO_EXPOSE_WORKLOADS/function
```

File: docs/03-tutorials/00-api-exposure/apix-10-expose-and-secure-workload-with-certificate.md [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+  <summary>
+  Call the secured endpoints of a service
+  </summary>
+
+Send a `GET` request to the HttpBin service with the client certificates that you used to create mTLS Gateway:
+
+   ```shell
+   curl --key ${CLIENT_CERT_KEY_FILE} \
+        --cert ${CLIENT_CERT_CRT_FILE} \
+        --cacert ${CLIENT_ROOT_CA_CRT_FILE} \ -ik -X GET https://httpbin.$DOMAIN_TO_EXPOSE_WORKLOADS/headers
```
Comment (videlov) [inline docs/03-tutorials/00-api-exposure/apix-10-expose-and-secure-workload-with-certificate.md:149]:
```suggestion
        --cacert ${CLIENT_ROOT_CA_CRT_FILE} \
        -ik -X GET https://httpbin-vs.$DOMAIN_TO_EXPOSE_WORKLOADS/headers
```

**Review by videlov — DISMISSED** [review]
/lgtm

File: docs/03-tutorials/00-api-exposure/apix-10-expose-and-secure-workload-with-certificate.md [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+
+This tutorial shows how to expose and secure a workload with mutual authentication using Kyma's mutual TLS Gateway.
+
+## Prerequisites
+
+This tutorial is based on a sample HttpBin service deployment and a sample Function. To deploy or create one of those, follow the [Create a workload](./apix-01-create-workload.md) tutorial.
+
+Before you start, Set up [mTLS Gateway](../00-security/sec-03-setup-mtls-gateway.md) to allow mutual authentication in Kyma and make sure that you exported the [bundle certificates](../00-security/sec-03-setup-mtls-gateway#steps).
+
+Optionally you can take a look on tutorial [How to create own self-signed Client Root CA and Certificate](../00-security/sec-02-mtls-selfsign-client-certicate.md)
```
Comment (grego952) [inline docs/03-tutorials/00-api-exposure/apix-10-expose-and-secure-workload-with-certificate.md:13]:
```suggestion
Optionally, take a look at the [How to create own self-signed Client Root CA and Certificate](../00-security/sec-02-mtls-selfsign-client-certicate.md) tutorial.
```

File: docs/03-tutorials/00-api-exposure/apix-10-expose-and-secure-workload-with-certificate.md [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+
+## Authorize client with a certificate
+
+Follow the instructions in the tabs to further secure the mTLS service or Function. Create AuthorizationPolicy that checks if the client's Common Name in the certificate matches.
```
Comment (grego952) [inline docs/03-tutorials/00-api-exposure/apix-10-expose-and-secure-workload-with-certificate.md:17]:
```suggestion
The following instructions describe how to further secure the mTLS service or Function. 
>**NOTE:** Create AuthorizationPolicy to check if the client's common name in the certificate matches.
```
Can you add some more information? With what`the client's common name in the certificate matches`?

File: docs/03-tutorials/00-security/sec-02-mtls-selfsign-client-certicate.md [hunk 9]
```diff
@@ -0,0 +1,44 @@
+---
+title: Prepare self-sign client certificates
```
Comment (grego952) [inline docs/03-tutorials/00-security/sec-02-mtls-selfsign-client-certicate.md:2]:
```suggestion
title: Prepare self-signed client certificates
```

File: docs/03-tutorials/00-security/sec-02-mtls-selfsign-client-certicate.md [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+---
+
+This tutorial shows how to create self-sign Root CA and use it to sign Client Certificate
```
Comment (grego952) [inline docs/03-tutorials/00-security/sec-02-mtls-selfsign-client-certicate.md:5]:
```suggestion
This tutorial shows how to create a self-signed root certificate authority (CA) and use it to sign a client certificate.
```

File: docs/03-tutorials/00-security/sec-02-mtls-selfsign-client-certicate.md [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+
+## Prepare a Client Root CA
```
Comment (grego952) [inline docs/03-tutorials/00-security/sec-02-mtls-selfsign-client-certicate.md:7]:
```suggestion
## Prepare a client root CA
```

File: docs/03-tutorials/00-security/sec-02-mtls-selfsign-client-certicate.md [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+1. Export the following values as environment variables:
+
+   ```bash
+   export CLIENT_ROOT_CA_CN={ROOT_CA_COMMON_NAME}
+   export CLIENT_ROOT_CA_ORG={ROOT_CA_ORGANIZATION}
+   export CLIENT_ROOT_CA_KEY_FILE=${CLIENT_ROOT_CA_CN}.key
+   export CLIENT_ROOT_CA_CRT_FILE=${CLIENT_ROOT_CA_CN}.crt
+   ```
+
+2. Generate a Client Root CA and a Client certificate:
```
Comment (grego952) [inline docs/03-tutorials/00-security/sec-02-mtls-selfsign-client-certicate.md:18]:
```suggestion
2. Generate a client root CA and a client certificate:
```

File: docs/03-tutorials/00-security/sec-02-mtls-selfsign-client-certicate.md [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+
+   ```bash
+   # Create a Root CA key and a certificate that's valid for one year - you can use it for validation
```
Comment (grego952) [inline docs/03-tutorials/00-security/sec-02-mtls-selfsign-client-certicate.md:21]:
```suggestion
   # Create a root CA key and a certificate that's valid for one year - you can use it for validation
```

File: docs/03-tutorials/00-security/sec-02-mtls-selfsign-client-certicate.md [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+   openssl req -x509 -sha256 -nodes -days 365 -newkey rsa:2048 -subj '/O=${CLIENT_ROOT_CA_ORG}/CN=${CLIENT_ROOT_CA_CN}' -keyout ${CLIENT_ROOT_CA_KEY_FILE} -out ${CLIENT_ROOT_CA_CRT_FILE}
+   ```
+   
+## Prepare Client Certificate
```
Comment (grego952) [inline docs/03-tutorials/00-security/sec-02-mtls-selfsign-client-certicate.md:25]:
```suggestion
## Prepare client certificate
```

File: docs/03-tutorials/00-security/sec-02-mtls-selfsign-client-certicate.md [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+
+   ```bash
+   export CLIENT_CERT_CN={COMMON_NAME}
+   export CLIENT_CERT_ORG={ORGANIZATION}
+   export CLIENT_CERT_CRT_FILE=${CLIENT_CERT_CN}.crt
+   export CLIENT_CERT_CSR_FILE=${CLIENT_CERT_CN}.csr
+   export CLIENT_CERT_KEY_FILE=${CLIENT_CERT_CN}.key
+   ```
+
+2. Generate a Client certificate and sign it with the Root CA:
```
Comment (grego952) [inline docs/03-tutorials/00-security/sec-02-mtls-selfsign-client-certicate.md:37]:
```suggestion
2. Generate a client certificate and sign it with the root CA:
```

File: docs/03-tutorials/00-security/sec-03-setup-mtls-gateway.md [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+2. Create a Namespace and export its value as an environment variable. Run:
+   >**NOTE:** Skip this step if you already have a Namespace
+
+      ```bash
+   export NAMESPACE={NAMESPACE_NAME}
+   kubectl create ns $NAMESPACE
+   kubectl label namespace $NAMESPACE istio-injection=enabled --overwrite
+   ```
+
+3. Export Gateway name, tls secret name and Client Root CA Cert File path
```
Comment (grego952) [inline docs/03-tutorials/00-security/sec-03-setup-mtls-gateway.md:29]:
```suggestion
3. Export Gateway name, TLS secret name, and client root Certificate Authority (CA) crt file path:
```

File: docs/03-tutorials/00-security/sec-03-setup-mtls-gateway.md [hunk 17]
```diff
@@ -0,0 +1,94 @@
...
+
+    ```bash
+   export MTLS_GATEWAY_NAME=mtls-gateway
+   export TLS_SECRET={TLS_SECRET_NAME} # The name of the TLS Secret that was created during the setup of the custom domain, for example, httpbin-tls-credentials
+   export CLIENT_ROOT_CA_CRT_FILE={CLIENT_ROOT_CA_CRT_FILE}
+   ```
+
+4. Create mTLS Gateway (mode: MUTUAL)
```
Comment (grego952) [inline docs/03-tutorials/00-security/sec-03-setup-mtls-gateway.md:37]:
```suggestion
4. Create mTLS Gateway (mode: MUTUAL):
```

File: docs/03-tutorials/00-security/sec-03-setup-mtls-gateway.md [hunk 18]
```diff
@@ -0,0 +1,94 @@
...
+           name: http
+           protocol: HTTP
+         tls:
+           httpsRedirect: true
+         hosts:
+           - '*.${DOMAIN_TO_EXPOSE_WORKLOADS}'
+   EOF
+   ```
+
+5. Add Client Root CA to cacert bundle secret for mTLS Gateway. Export the following value as an environment variable and run the command provided:
```
Comment (grego952) [inline docs/03-tutorials/00-security/sec-03-setup-mtls-gateway.md:78]:
```suggestion
5. Add client root CA to CA cert bundle secret for mTLS Gateway. Export the following value as an environment variable and run the command:
```

**Review by triffer — APPROVED** [review]

**Review by werdes72 — APPROVED** [review]

**Review by grego952 — APPROVED** [review]

File: docs/04-operation-guides/security/sec-07-useful-links.md [hunk 19]
```diff
@@ -5,7 +5,7 @@ title: Useful links
 If you're interested in learning more about Security in Kyma, follow these links to:
 
 - Perform some simple and more advanced tasks:
-  - [Update the TLS certificate](../../03-tutorials/sec-01-tls-certificates-security.md)
+  - [Update the TLS certificate](../../03-tutorials/00-security/sec-01-tls-certificates-security.md)
```
Comment (strekm) [inline docs/04-operation-guides/security/sec-07-useful-links.md:8]: please add other tutorials

**Pull request merged**